File: calculator/app/src/main/java/com/example/calculator/MainActivity.java
package com.example.calculator;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.google.android.material.button.MaterialButton;
import org.mozilla.javascript.Context;
import org.mozilla.javascript.Scriptable;

public class MainActivity extends cal implements View.OnClickListener {
    TextView resulttv ;
    TextView solutiontv;
    MaterialButton btnc,btnopnbrc,btnclsbrc,btndiv,btnminus,btnplus,btnx,btnequal,btnmod;
    MaterialButton btn7,btn8,btn9,btn6,btn5,btn3,btn4,btn2,btn1,btn0,btndot;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        resulttv=findViewById(R.id.resulttv);
        solutiontv=findViewById(R.id.solutiontv);

        assingnId(btn0,R.id.btn0);
        assingnId(btn1,R.id.btn1);
        assingnId(btn2,R.id.btn2);
        assingnId(btn3,R.id.btn3);
        assingnId(btn4,R.id.btn4);
        assingnId(btn5,R.id.btn5);
        assingnId(btn6,R.id.btn6);
        assingnId(btn7,R.id.btn7);
        assingnId(btn8,R.id.btn8);
        assingnId(btn9,R.id.btn9);
        assingnId(btnplus,R.id.btnplus);
        assingnId(btnminus,R.id.btnminus);
        assingnId(btndiv,R.id.btndiv);
        assingnId(btnx,R.id.btnx);
        assingnId(btndot,R.id.btndot);
        assingnId(btnc,R.id.btnc);
        assingnId(btnequal,R.id.btnequal);
        assingnId(btnopnbrc,R.id.btnopnbrc);
        assingnId(btnclsbrc,R.id.btnclsbrc);
        assingnId(btnmod,R.id.btnmod);



    } void assingnId(MaterialButton btn,int id){
        btn=findViewById(id);
        btn.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        MaterialButton button = (MaterialButton) view;
        String buttonText = button.getText().toString();
        String dataToCalculate = solutiontv.getText().toString();


        if (buttonText.equals("=")) {
            solutiontv.setText(resulttv.getText());
            return;
        }
        if(buttonText.equals("AC")) {
        solutiontv.setText("");
        resulttv.setText("0");
        return;
        }
        if (buttonText.equals("C")){
            dataToCalculate=dataToCalculate.substring(0,dataToCalculate.length()-1);
        }
      else{dataToCalculate = dataToCalculate + buttonText;
       }

        solutiontv.setText(dataToCalculate);
        String finalresult=getResult(dataToCalculate);

        if(!finalresult.equals("Error")){
            resulttv.setText(finalresult);
        }
    }
    String getResult(String data){
        try {
          Context context=Context.enter();
          context.setOptimizationLevel(-1);
          Scriptable scriptable= context.initStandardObjects();
         String finalResult= context.evaluateString(scriptable,data,"javascript",1,null).toString();
         if(finalResult.endsWith(".0")) {
         finalResult=finalResult.replace(".0","");
         }
          return finalResult;
        }catch (Exception a){
            return "Error";
        }
}}

File: task1/app/src/main/java/com/example/task1/MainActivity.kt
package com.example.task1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button


class MainActivity : AppCompatActivity(),View.OnClickListener{
    private var a = arrayListOf<String>("One","Two","Three","Four","Five")

    var btn1:Button? =null
    var btn2:Button?=null
    var btn3:Button? =null
    var btn4:Button? =null
    var btn5:Button?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn1 = findViewById(R.id.btn1)
        btn2 = findViewById(R.id.btn2)
        btn3 = findViewById(R.id.btn3)
        btn4 = findViewById(R.id.btn4)
        btn5 = findViewById(R.id.btn5)
        (btn1)?.setOnClickListener(this)
        (btn2)?.setOnClickListener(this)
        (btn3)?.setOnClickListener(this)
        (btn4)?.setOnClickListener(this)
        (btn5)?.setOnClickListener(this)

        btn1?.text = a.get(3)
        btn2?.text = a.get(4)
        btn3?.text = a.get(0)
        btn4?.text = a.get(2)
        btn5?.text = a.get(1)

    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.btn1 -> {
                btn1?.text=a.get(0)
                btn2?.text=a.get(1)
                btn3?.text=a.get(2)
                btn4?.text=a.get(3)
                btn5?.text=a.get(4)
            }
            R.id.btn2 -> {
                btn1?.text=a.get(4)
                btn2?.text=a.get(0)
                btn3?.text=a.get(1)
                btn4?.text=a.get(2)
                btn5?.text=a.get(3)
            }
            R.id.btn3 -> {
                btn1?.text=a.get(3)
                btn2?.text=a.get(4)
                btn3?.text=a.get(0)
                btn4?.text=a.get(1)
                btn5?.text=a.get(2)
            }
            R.id.btn4 -> {
                btn1?.text=a.get(2)
                btn2?.text=a.get(3)
                btn3?.text=a.get(4)
                btn4?.text=a.get(0)
                btn5?.text=a.get(1)
            }
            R.id.btn5 -> {
                btn1?.text=a.get(1)
                btn2?.text=a.get(2)
                btn3?.text=a.get(3)
                btn4?.text=a.get(4)
                btn5?.text=a.get(0)
            }
        }


    }}








